lxd/storage_zfs.go

```diff
@@ -87,24 +87,32 @@ func (s *storageZfs) StoragePoolCheck() error {
 	}
 
 	poolName := s.getOnDiskPoolName()
-	if filepath.IsAbs(source) {
-		if zfsFilesystemEntityExists(poolName, "") {
-			return nil
-		}
-		logger.Debugf("ZFS storage pool \"%s\" does not exist. Trying to import it.", poolName)
+	exists := zfsFilesystemEntityExists(poolName, "")
+	if exists {
+		return nil
+	}
+
+	logger.Debugf("ZFS storage pool \"%s\" does not exist. Trying to import it.", poolName)
 
+	var err error
+	var msg string
+	if filepath.IsAbs(source) {
 		disksPath := shared.VarPath("disks")
-		output, err := shared.RunCommand(
-			"zpool",
-			"import",
-			"-d", disksPath, poolName)
-		if err != nil {
-			return fmt.Errorf("ZFS storage pool \"%s\" could not be imported: %s", poolName, output)
+		msg, err = shared.RunCommand("zpool", "import", "-d", disksPath, poolName)
+	} else {
+		purePoolName := poolName
+		idx := strings.Index(poolName, "/")
+		if idx != -1 {
+			purePoolName = poolName[:idx]
```

Comment:
That looks more complicated than it should be, don't you just want: strings.Split(poolName, "/")[0] here?

Reply:
@brauner saw that one?

Reply:
Oh weird, the connection must have broken off when I pushed this on the plane.

Reply:
done